extensions/azcli/src/az.ts

```diff
@@ -508,6 +509,9 @@ async function findSpecificAzAndArc(): Promise<IAzTool> {
 		throw AzureCLIArcExtError;
 	}
 
+	// Quietly attempt to update the arcdata extension to the latest. If it is already the latest, then it will not update.
```

Comment:
Why quietly? Generally this kind of thing is best done as a prompt, users tend to not like surprise installations.

Reply:
I argued this as well, but the reason why we ended up deciding to quietly update the extension was that if users declined an update, features in ADS would not work as expected and users may be left wondering why. During shiproom we did acknowledge that if customers start asking us why their extensions are being updated automatically, we could make it a prompt.